### PR TITLE
[SDK] Add winnt_old entry for ARM64 and start fixing ARM64 PE binaries 

### PR DESCRIPTION
**Pull request by DarkFire01**
## Purpose

- Grabbed definitions based on the latest SDK for arm64
- Added entries for ARM64 into pefixup and the geninc tool

### CODE REVIEW

File: sdk/include/xdk/winnt_old.h [hunk 1]
```diff
@@ -2148,7 +2148,150 @@ typedef struct _DISPATCHER_CONTEXT
     PBYTE  NonVolatileRegisters;
     DWORD Reserved;
 } DISPATCHER_CONTEXT, *PDISPATCHER_CONTEXT;
+#elif defined(_ARM64_)
```
Comment (binarymaster) [inline sdk/include/xdk/winnt_old.h:2151]:
```suggestion
#elif defined(_ARM64_)

```

File: sdk/tools/CMakeLists.txt [hunk 2]
```diff
@@ -50,7 +50,7 @@ if(NOT MSVC)
     add_subdirectory(rsym)
 
     add_host_tool(pefixup pefixup.c)
-if (ARCH STREQUAL "amd64")
+if ((ARCH STREQUAL "amd64") OR (ARCH STREQUAL "arm64"))
```
Comment (binarymaster) [inline sdk/tools/CMakeLists.txt:53]:
```suggestion
if (ARCH STREQUAL "amd64" OR ARCH STREQUAL "arm64")
```

File: sdk/tools/geninc/geninc.c [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #define IMAGE_FILE_MACHINE_I386  0x014c
 #define IMAGE_FILE_MACHINE_AMD64 0x8664
 #define IMAGE_FILE_MACHINE_ARMNT 0x01c4
+#define IMAGE_FILE_MACHINE_ARM64 0xAA64
```
Comment (binarymaster) [inline sdk/tools/geninc/geninc.c:13]:
```suggestion
#define IMAGE_FILE_MACHINE_ARM64 0xaa64
```

**Review by binarymaster — COMMENTED** [review]
Some suggestions

File: sdk/include/xdk/winnt_old.h [hunk 4]
```diff
@@ -2148,7 +2148,151 @@ typedef struct _DISPATCHER_CONTEXT
...
+    //
+
+    /* +0x000 */ DWORD ContextFlags;
+
+    //
+    // Integer registers
+    //
+
+    /* +0x004 */ DWORD Cpsr;       // NZVF + DAIF + CurrentEL + SPSel
+    /* +0x008 */ union {
```
Comment (HBelusca) [inline sdk/include/xdk/winnt_old.h:2193]: Indentation ugly; can we have the offsets on the right instead?

Comment (DarkFire01) [reply sdk/include/xdk/winnt_old.h:2193]: This is just how offsets were in the SDK, We can change it if you guys would like

Comment (HBelusca) [reply sdk/include/xdk/winnt_old.h:2193]: hm ok, if this is a direct copy-paste of their SDK then I understand (but we would like not direct copypaste of MS SDK code either).

Comment (DarkFire01) [reply sdk/include/xdk/winnt_old.h:2193]: Sounds good I'll change it, I wanted to change the type definitions to be like the rest of our headers anyway so this works out good!

**Review by binarymaster — APPROVED** [review]

**Pull request merged**